src/Filters/TimeAgoFilter.php

```diff
@@ -46,12 +46,33 @@ public function timeAgo($datetime)
                 continue;
             }
 
-            $transString = 'diff.'.($isFuture ? 'in.' : 'ago.').$val;
+            $dayDiff = $this->getDays($datetime);
+
+            $transString = ($dayDiff !== false) ? $dayDiff : 'diff.'.($isFuture ? 'in.' : 'ago.').$val;
+
             $numberOfUnits = floor($time / $unit);
 
             $output = $this->translator->transChoice($transString, $numberOfUnits, ['%count%' => $numberOfUnits]);
 
             return $output;
         }
     }
+
+    protected function getDays($date)
+    {
+        $dateDetails = date_parse($date);
+        if ($date == date('Y-m-d', strtotime(('-1 day'))) && $dateDetails['hour'] == false) {
```

Comment:
Switch the conditions around. It won't parse the date and do the math if the first condition is false, which makes it more performant.